### DISCUSSION TITLE
(curious) How are the threshold values being determined?

### DISCUSSION THREAD

**Discussion by szelok** [aurelio-labs/semantic-router#61] [General]
https://github.com/aurelio-labs/semantic-router/blob/main/semantic_router/layer.py#L153C1-L158C40  
```
      if isinstance(encoder, OpenAIEncoder):
            self.score_threshold = 0.82
        elif isinstance(encoder, CohereEncoder):
            self.score_threshold = 0.3
        else:
            self.score_threshold = 0.82

```

**Comment by jamescalam** [comment]
depends on the route layer, but the default `RouteLayer` is using semantic similarity followed by aggregation and filtering using the `score_threshold` values you showed above

**Comment by szelok** [comment]
@jamescalam - my question was how we should determine the threshold score. For example, you set the threshold to 0.82 for OpenAI but 0.3 for Cohere (a big difference). Any advice?

Also, it feels like it requires good 'utterance' engineering. For example I tried asking 'What do you think about Joe Biden' in your example (fro HuggingFaceEncoder). It didn't match any routes. Just my two cents.
